static/app/components/events/interfaces/frame/stacktraceLink.tsx

```diff
@@ -21,187 +25,209 @@ import type {
 } from 'sentry/types';
 import {StacktraceLinkEvents} from 'sentry/utils/analytics/integrations/stacktraceLinkAnalyticsEvents';
 import {getAnalyicsDataForEvent} from 'sentry/utils/events';
-import handleXhrErrorResponse from 'sentry/utils/handleXhrErrorResponse';
 import {
   getIntegrationIcon,
   trackIntegrationAnalytics,
 } from 'sentry/utils/integrationUtil';
 import {promptIsDismissed} from 'sentry/utils/promptIsDismissed';
-import withOrganization from 'sentry/utils/withOrganization';
-import withProjects from 'sentry/utils/withProjects';
+import {useQuery, useQueryClient} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
 
 import {OpenInContainer} from './openInContextLine';
 import StacktraceLinkModal from './stacktraceLinkModal';
 
-type Props = AsyncComponent['props'] & {
+interface StacktraceLinkSetupProps {
   event: Event;
-  frame: Frame;
-  line: string;
-  lineNo: number;
   organization: Organization;
-  projects: Project[];
-};
-
-type State = AsyncComponent['state'] & {
-  isDismissed: boolean;
-  match: StacktraceLinkResult;
-  promptLoaded: boolean;
-};
-
-class StacktraceLink extends AsyncComponent<Props, State> {
-  get project() {
-    // we can't use the withProject HoC on an the issue page
-    // so we ge around that by using the withProjects HoC
-    // and look up the project from the list
-    const {projects, event} = this.props;
-    return projects.find(project => project.id === event.projectID);
-  }
-
-  componentDidMount() {
-    this.promptsCheck();
-  }
+  project?: Project;
+}
 
-  async promptsCheck() {
-    const {organization} = this.props;
+function StacktraceLinkSetup({organization, project, event}: StacktraceLinkSetupProps) {
+  const api = useApi();
+  const [promptDismissed, setPromptDismissed] = useState(false);
```

Comment:
Do we need this `useState` anymore? Seems like we can get this from the query cache like you did below:

```
const isPromptDismissed =
    prompt.isSuccess && prompt.data.data
      ? promptIsDismissed({
          dismissedTime: prompt.data.data.dismissed_ts,
          snoozedTime: prompt.data.data.snoozed_ts,
        })
      : false;
```